src/test/java/com/amazonaws/encryptionsdk/TestVectorRunner.java

```diff
@@ -126,21 +139,15 @@ private static InputStream readFromJar(JarFile jar, String name) throws IOExcept
         return jar.getInputStream(entry);
     }
 
-    private static void cacheData(JarFile jar, String url) throws IOException {
-        if (!cachedData.containsKey(url)) {
-            cachedData.put(url, readBytesFromJar(jar, url));
-        }
-    }
-
     @SuppressWarnings("unchecked")
     private static TestCase parseTest(String testName, Map<String, Object> data, Map<String, KeyEntry> keys,
-                                      JarFile jar, KmsMasterKeyProvider kmsProv) throws IOException {
+                                      JarFile jar) {
         final String plaintextUrl = (String) data.get("plaintext");
-        cacheData(jar, plaintextUrl);
         final String ciphertextURL = (String) data.get("ciphertext");
-        cacheData(jar, ciphertextURL);
+        cachedData.putIfAbsent(plaintextUrl, readBytesFromJar(jar, plaintextUrl));
+        cachedData.putIfAbsent(ciphertextURL, readBytesFromJar(jar, ciphertextURL));
```

Comment:
This always reads the data from the JAR rather than only reading it if there is a cache miss. That might be expensive as I'm not sure how the data is being cached locally by Java. You could use `computeIfAbsent` if you want though.

Reply:
Fixed